test/main_test.go

```
@@ -0,0 +1,158 @@
+		"bool_a":            []interface{}{false, true, false},
+		"bytes_a":           []interface{}{"TestBytes1", "TestBytes2"},
+		"timestamp_a":       []interface{}{defaultTimestamp},
+		"duration_custom_a": []interface{}{"1m"},
+
+		"nested": []interface{}{
+			map[string]interface{}{
+				"str": "TestString",
+				"nested": []interface{}{
+					map[string]interface{}{
+						"str": "TestString1",
+					},
+					map[string]interface{}{
+						"str": "TestString2",
+					},
+				},
+				"nested_m": map[string]interface{}{
+					"kn1": "vn1",
+					"kn2": "vn2",
+				},
+			},
+		},
+
+		"nested_m": map[string]interface{}{
+			"k1": "v1",
+			"k2": "v2",
+		},
+
+		"nested_m_obj": []interface{}{
+			map[string]interface{}{
+				"key": "obj1",
+				"value": []interface{}{
+					map[string]interface{}{
+						"str": "TestString1",
+					},
+				},
+			},
+		},
+	}
+)
+
+// buildSubject builds Test struct from test fixture data
+func buildSubject(t *testing.T) (*Test, error) {
+	subject := &Test{}
+	data := schema.TestResourceDataRaw(t, SchemaTest(), fixutre)
+	err := UnmarshalTest(data, subject)
+	return subject, err
+}
+
+// TestUnmarshal ensures that parsing is done without errors
+func TestUnmarshal(t *testing.T) {
+	_, err := buildSubject(t)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// TestElementaries ensures decoding of elementary types
+func TestElementaries(t *testing.T) {
+	subject, _ := buildSubject(t)
+
+	assert.Equal(t, subject.Str, "TestString", "Test.Str")
+	assert.Equal(t, subject.Int32, int32(999), "Test.Int32")
+	assert.Equal(t, subject.Int64, int64(998), "Test.Int64")
+	assert.Equal(t, subject.Float, float32(18.1), "Test.Float")
+	assert.Equal(t, subject.Double, float64(18.4), "Test.Dobule")
+	assert.Equal(t, subject.Bool, true, "Test.Bool")
+	assert.Equal(t, subject.Bytes, []byte("TestBytes"), "Test.Bytes")
+}
+
+// TestTimes ensures decoding of time and duration fields
+func TestTimes(t *testing.T) {
+	subject, _ := buildSubject(t)
```

@gzigzigzeo here and below, let's never skip or ignore any errors.

---

Yep, I think that here and in previous step, I tried to unnecessarily DRY tests. So, instead of panicking, I would better check this error every time with require package and eliminate TestUnmarshal method.